core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -90,9 +106,51 @@ public RestStatus getStatus() {
             return this.status;
         }
 
+        /**
+         * The actual cause of the failure.
+         */
         public Throwable getCause() {
             return cause;
         }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            index = in.readString();
+            type = in.readString();
+            id = in.readOptionalString();
+            cause = in.readThrowable();
+            status = ExceptionsHelper.status(cause);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(getIndex());
+            out.writeString(getType());
+            out.writeOptionalString(getId());
+            out.writeThrowable(getCause());
+        }
+
+        static final class Fields {
+            static final XContentBuilderString INDEX = new XContentBuilderString("index");
+            static final XContentBuilderString TYPE = new XContentBuilderString("type");
+            static final XContentBuilderString ID = new XContentBuilderString("id");
+            static final XContentBuilderString CAUSE = new XContentBuilderString("cause");
+            static final XContentBuilderString STATUS = new XContentBuilderString("status");
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(INDEX, index);
+            builder.field(TYPE, type);
+            if (id != null) {
+                builder.field(ID, id);
+            }
+            builder.startObject(CAUSE);
+            ElasticsearchException.toXContent(builder, params, cause);
+            builder.endObject();
+            builder.field(STATUS, status);
```

Comment:
this should be `status.getStatus()`. Otherwise, users get the text instead of the HTTP status code. See also commit f5d307e. 

Here is a snippet from an example response based on the current code:

```
{
         "index": "movies_ext",
         "type": "movie",
         "id": "2",
         "cause": {
            "type": "unavailable_shards_exception",
            "reason": "[movies_ext][2] Not enough active copies to meet write consistency of [ALL] (have 1, needed 2). Timeout: [1m], request: org.elasticsearch.action.bulk.BulkShardRequest@541ca17d"
         },
         "status": "SERVICE_UNAVAILABLE"
      }
```

Expected behavior: status should be "503", not "SERVICE_UNAVAILABLE".